change to new C++ interface way

AlwinEsch opened the pull request:
Related to xbmc/xbmc#12336

Also becomes the use of timidity changed. There is it now a shared library
who becomes on start copied to another name to allow open with dlopen.
This way allow multiple instances and Kodi no more need itself the hack way.

@notspiff don't shot me :-D

Has tested this way and on Linux it works good, for other OS must it be also checked. But if acceptable for you can be from Kodi itself the way with lib copy removed and be done from addon if needed.

This allow also for audio decoder to use in multiple instance way.

AlwinEsch committed:
change settings.xml to new way

notspiff commented:
Please separate the child stuff functionality into some base class. Then we can stick this in kodiplatform and reuse is as simple as earlier while nastyness is outside the interfaces.

AlwinEsch commented:
Have added something, not sure if it is what you mean but make it easier and much smaller.

The "DllHelper.h" can be added to kodi-platform or to the dev-kit headers (normally I want to have everything inside them ;-) )

notspiff commented:
Nice

AlwinEsch committed:
add .gitignore

AlwinEsch commented:
Is updated, and the Dll Helper now included in Kodi.

Have you the possibility to check the build on Windows?

AlwinEsch commented:
Have a question about audiodecoder.vgmstream. Where I can find file examples to test them? Has googled and found a lot, but nowhere a test file to the extensions from them :-) .

AlwinEsch commented:
Ping @notspiff 
This and the other can in and have you same example files for audiodecoder.vgmstream.

AlwinEsch commented:
Ups, this here can in if working is confirmed on other OS.

AlwinEsch committed:
change to new C++ interface way

Also becomes the use of timidity changed. There is it now a shared library
who becomes on start copied to another name to allow open with dlopen.
This way allow multiple insatnces and Kodi no more need itself the hack way.

notspiff commented:
it's closer to what it should be in any case, on all os'es.

notspiff commented:
for vgmstream use https://www.zophar.net/music/nintendo-gamecube-gcn (the non-mp3 downloads obviously).

notspiff commented:
btw i just fixed a couple of bugs so now it works for adp files.
